test/Test/BaseDAO.hs

```diff
@@ -0,0 +1,166 @@
+          }
+      ]
+
+-- TODO: Replace SampleFA2 with BaseDAO
+baseDAOContract :: FA2.PermissionsDescriptor -> Contract FA2.Parameter SampleFA2.Storage
+baseDAOContract p =
+  SampleFA2.fa2Contract p
+
+originateBaseDAO
+  :: MonadNettest caps base m => FA2.PermissionsDescriptor
+  -> m ((Address, Address), (Address, Address), TAddress FA2.Parameter)
+originateBaseDAO p = do
+  owner1  :: Address <- newAddress "owner1"
+  operator1  :: Address <- newAddress "operator1"
+  owner2  :: Address <- newAddress "owner2"
+  operator2  :: Address <- newAddress "operator2"
+
+  let bal = Map.fromList
+        [ ((owner1, 0), 100)
+        , ((owner2, 0), 100)
+        ]
+  let (operators :: SampleFA2.Operators) =
+        BigMap $ Map.fromList
+          [ ((owner1, operator1), ())
+          , ((owner2, operator2), ())
+          ]
+  let
+    originateData = OriginateData
+      { odFrom = AddressResolved genesisAddress
+      , odName = "BaseDAO"
+      , odBalance = toMutez 0
+      , odStorage = baseDAOStorage bal operators
+      , odContract = baseDAOContract p
+      }
+  dao <- originate originateData
+  pure ((owner1, operator1), (owner2, operator2), dao)
+
+test_BaseDAO :: TestTree
+test_BaseDAO = testGroup "tests to check BaseDAO contract functionality"
+  [ testCase "Allows zero transfer from non-existent accounts" $
+      nettestTestExpectation zeroTransferScenario
+  , testCase "Is denied if only owner transfer is allowed" $
+      nettestTestExpectation ownerTransferAllowScenario
+  , testCase "aborts if there is a failure (due to low balance)" $
+      nettestTestExpectation lowBalanceScenario
+  -- TODO: testcase: "Allows zero transfer from non-existent accounts"
+  -- TODO: testcase: "Is denied if only owner transfer is allowed"
+  -- TODO: testcase: "aborts if there is a failure (due to low balance)"
+  -- TODO: testcase: "aborts if there is a failure (due to non existent source account)"
+  -- TODO: testcase: "aborts if there is a failure (due to bad operator)"
+  -- TODO: testcase: "aborts if there is a failure (due to bad operator for zero transfer)"
+  -- TODO: testcase: "accepts an empty list of transfers"
+  -- TODO: testcase: "is denied if operator transfer is denied in permissions descriptior"
+  -- TODO: testcase: "validates token id"
+  -- TODO: testcase: "cannot transfer foreign money"
+
+  -- Self Transfer
+  -- TODO: testcase: "Cannot transfer foreign money"
+  -- TODO: testcase: "is denied if self transfer is forbidden in permissions descriptior"
+  -- TODO: testcase: "validates token id"
+  -- TODO: testcase: "aborts if there is a failure (due to low balance)"
+  -- TODO: testcase: "aborts if there is a failure (due to non existent source account)"
+  -- TODO: testcase: "aborts if there is a failure (due to bad token id)"
+  -- TODO: testcase: "balanceOf request validates token id"
+  -- TODO: testcase: "balanceOf request accepts empty list"
+  -- TODO: testcase: "Permissions_descriptor entrypoint is available"
```

Comment:
We don't need this entrypoint since we are using the default permission descriptor. I am not going to continue, it looks like at least one more test (the next one) is redundant. Also passing the descriptor to `baseDAOContract` is redundant.